pkg/storage/storage.go

```diff
@@ -120,11 +134,13 @@ func (s *storageModule) initialize(policy pkg.StoragePolicy) error {
 		}
 		_, err = volume.Mount()
 		if err != nil {
-			return err
+			s.brokenPools = append(s.brokenPools, pkg.BrokenPool{Label: volume.Name(), Err: err})
+			log.Warn().Msgf("Failed to mount volume %v", volume.Name())
+			continue
 		}
 		log.Debug().Msgf("Mounted volume %s", volume.Name())
+		s.volumes = append(s.volumes, volume)
 	}
-	s.volumes = append(s.volumes, existingPools...)
 
 	// list disks
 	log.Info().Msgf("Finding free disks")
```

Comment:
Can't you also have some error when listing the devices ?

Reply:
I am not intimately familiar with the way lsblk returns it outputs, however I think that it either detects the disk, appending it to the returned disks, or does not detect the disk, and ignores it. Errors returned here are execution errors for the lsblk invocation, and should not be related to any malfunctioning disks afaik. Please let me know if I got this wrong